__tests__/app.test.js

```diff
@@ -0,0 +1,193 @@
+const { getTopics } = require("../db/helpers/utils");
+
+const request = require("supertest");
+const app = require("../app");
+const db = require("../db/connection");
+const seed = require("../db/seeds/seed");
+const testData = require("../db/data/test-data/index");
+
+beforeEach(() => {
+  return seed(testData);
+});
+
+afterAll(() => {
+  return db.end();
+});
+
+describe("getTopics", () => {
+  test("Responds with an array of topic objects, slug & description", () => {
+    return request(app)
+      .get("/api/topics")
+      .expect(200)
+      .then((response) => {
+        expect(response.body.topics).toEqual([
+          {
+            description: "The man, the Mitch, the legend",
+            slug: "mitch",
+          },
+          {
+            description: "Not dogs",
+            slug: "cats",
+          },
+          {
+            description: "what books are made of",
+            slug: "paper",
+          },
+        ]);
+      });
+  });
+  test("should respond with 404 path not found", () => {
+    return request(app)
+      .get("/api/topicz")
+      .expect(404)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("404 - path not found");
+      });
+  });
+});
+
+describe("getArticles", () => {
+  test("Responds with the required first article", () => {
+    return request(app)
+      .get("/api/articles/1")
+      .expect(200)
+      .then((articles) => {
+        expect(articles.body).toEqual(
+          expect.objectContaining({
+            author: expect.any(String),
+            title: expect.any(String),
+            article_id: 1,
+            body: expect.any(String),
+            topic: expect.any(String),
+            created_at: expect.any(String),
+            votes: expect.any(Number),
+          })
+        );
+      });
+  });
```

Comment:
excellent test, this is what i would like your topics test to do! (with the added array length check)

Reply:
we do not need the body here, this will only be sent when we ask for the article by ID. you will need to amend the model as well.

Reply:
Fixed topics & added length.